### PR TITLE
Return LexErrors from `Lexer::next()` instead of CompileErrors

### PR DESCRIPTION
**Pull request by jyn514**

### CODE REVIEW

File: src/data/error.rs [hunk 1]
```diff
@@ -490,11 +510,6 @@ mod tests {
         );
     }
 
-    #[test]
-    fn test_compile_error_from_locatable_string() {
-        let _ = CompileError::from(Location::default().with("apples".to_string()));
```
Comment (pythondude325) [inline src/data/error.rs]: What was the reason to remove this test?

File: src/lex/cpp.rs [hunk 2]
```diff
@@ -210,7 +210,11 @@ impl<'a> PreProcessor<'a> {
     }
     #[inline]
     fn next_token(&mut self) -> Option<CppResult<Token>> {
-        self.lexer_mut().next()
+        match self.lexer_mut().next() {
+            Some(Err(err)) => Some(Err(err.into())),
+            Some(Ok(ok)) => Some(Ok(ok)),
+            None => None,
```
Comment (pythondude325) [inline src/lex/cpp.rs]: This could be replaced be some `map`s, do you prefer the `match` for some reason?

File: src/lex/mod.rs [hunk 3]
```diff
@@ -654,7 +656,7 @@ impl Lexer {
 impl Iterator for Lexer {
     // option: whether the stream is exhausted
     // result: whether the next lexeme is an error
-    type Item = CompileResult<Locatable<Token>>;
+    type Item = Result<Locatable<Token>, Locatable<LexError>>;
```
Comment (pythondude325) [inline src/lex/mod.rs]: Use the `LexResult` alias here.

**Comment by jyn514** [comment]
Addressed review comments.

**Pull request merged**